diff -dur -x '*~' Carla-1.9.6.orig/Makefile Carla-1.9.6/Makefile
--- Carla-1.9.6.orig/Makefile	2015-03-12 14:09:14.000000000 +0100
+++ Carla-1.9.6/Makefile	2016-11-19 13:47:16.395089673 +0100
@@ -398,9 +398,9 @@
 install:
 	# Create directories
 	install -d $(DESTDIR)$(PREFIX)/bin/
-	install -d $(DESTDIR)$(PREFIX)/lib/carla/
-	install -d $(DESTDIR)$(PREFIX)/lib/carla/styles/
-	install -d $(DESTDIR)$(PREFIX)/lib/pkgconfig/
+	install -d $(DESTDIR)$(LIBDIR)/carla/
+	install -d $(DESTDIR)$(LIBDIR)/carla/styles/
+	install -d $(DESTDIR)$(LIBDIR)/pkgconfig/
 	install -d $(DESTDIR)$(PREFIX)/include/carla/
 	install -d $(DESTDIR)$(PREFIX)/include/carla/includes/
 	install -d $(DESTDIR)$(PREFIX)/share/applications/
@@ -461,28 +461,28 @@
 	install -m 644 data/carla.xml $(DESTDIR)$(PREFIX)/share/mime/packages/
 
 	# Install pkg-config file
-	install -m 644 data/carla-standalone.pc $(DESTDIR)$(PREFIX)/lib/pkgconfig/
+	install -m 644 data/carla-standalone.pc $(DESTDIR)$(LIBDIR)/pkgconfig/
 
 	# Install backend libs
 	install -m 644 \
 		bin/libcarla_*.* \
-		$(DESTDIR)$(PREFIX)/lib/carla/
+		$(DESTDIR)$(LIBDIR)/carla/
 
 	# Install other binaries
 	install -m 755 \
 		bin/*bridge-* \
 		bin/carla-discovery-* \
-		$(DESTDIR)$(PREFIX)/lib/carla/
+		$(DESTDIR)$(LIBDIR)/carla/
 
 	# Install the real modgui bridge
 	install -m 755 \
 		data/carla-bridge-lv2-modgui \
-		$(DESTDIR)$(PREFIX)/lib/carla/
+		$(DESTDIR)$(LIBDIR)/carla/
 
 	# Install theme
 	install -m 644 \
 		bin/styles/* \
-		$(DESTDIR)$(PREFIX)/lib/carla/styles/
+		$(DESTDIR)$(LIBDIR)/carla/styles/
 
 	# Install python code
 	install -m 644 \
@@ -578,41 +578,43 @@
 	$(LINK) $(PREFIX)/share/carla/ui_inputdialog_value.py     $(DESTDIR)$(PREFIX)/share/carla/resources/
 
 	# Adjust PREFIX value in script files
-	sed -i "s?X-PREFIX-X?$(PREFIX)?" \
+	sed -i "s?X-PREFIX-X?$(PREFIX)?;s?X-LIBDIR-X?$(LIBDIR)?" \
+		$(DESTDIR)$(PREFIX)/share/carla/carla_shared.py \
+		$(DESTDIR)$(PREFIX)/share/carla/carla_widgets.py \
 		$(DESTDIR)$(PREFIX)/bin/carla \
 		$(DESTDIR)$(PREFIX)/bin/carla-database \
 		$(DESTDIR)$(PREFIX)/bin/carla-patchbay \
 		$(DESTDIR)$(PREFIX)/bin/carla-rack \
 		$(DESTDIR)$(PREFIX)/bin/carla-single \
 		$(DESTDIR)$(PREFIX)/bin/carla-settings \
-		$(DESTDIR)$(PREFIX)/lib/carla/carla-bridge-lv2-modgui \
-		$(DESTDIR)$(PREFIX)/lib/pkgconfig/carla-standalone.pc
+		$(DESTDIR)$(LIBDIR)/carla/carla-bridge-lv2-modgui \
+		$(DESTDIR)$(LIBDIR)/pkgconfig/carla-standalone.pc
 # 		$(DESTDIR)$(PREFIX)/bin/carla-control \
 
 	# --------------------------------------------------------------------------------------------------------------------
 
 	# Install lv2 plugin
-	install -d $(DESTDIR)$(PREFIX)/lib/lv2/carla.lv2/
+	install -d $(DESTDIR)$(LIBDIR)/lv2/carla.lv2/
 
 	install -m 644 \
 		bin/carla.lv2/carla.* \
 		bin/carla.lv2/*.ttl \
-		$(DESTDIR)$(PREFIX)/lib/lv2/carla.lv2/
+		$(DESTDIR)$(LIBDIR)/lv2/carla.lv2/
 
 	# Link binaries for lv2 plugin
-	@for i in $(shell find $(DESTDIR)$(PREFIX)/lib/carla/ -maxdepth 1 -type f -exec basename {} ';'); do \
-		$(LINK) $(PREFIX)/lib/carla/$$i $(DESTDIR)$(PREFIX)/lib/lv2/carla.lv2/; \
-		$(LINK) $(PREFIX)/lib/carla/$$i $(DESTDIR)$(PREFIX)/lib/vst/carla.vst/; \
+	@for i in $(shell find $(DESTDIR)$(LIBDIR)/carla/ -maxdepth 1 -type f -exec basename {} ';'); do \
+		$(LINK) $(LIBDIR)/carla/$$i $(DESTDIR)$(LIBDIR)/lv2/carla.lv2/; \
+		$(LINK) $(LIBDIR)/carla/$$i $(DESTDIR)$(LIBDIR)/vst/carla.vst/; \
 	done
-	rm -f $(DESTDIR)$(PREFIX)/lib/lv2/carla.lv2/libcarla_standalone2.*
-	rm -f $(DESTDIR)$(PREFIX)/lib/vst/carla.vst/libcarla_standalone2.*
+	rm -f $(DESTDIR)$(LIBDIR)/lv2/carla.lv2/libcarla_standalone2.*
+	rm -f $(DESTDIR)$(LIBDIR)/vst/carla.vst/libcarla_standalone2.*
 
 	# Link styles for lv2 plugin
-	$(LINK) $(PREFIX)/lib/carla/styles $(DESTDIR)$(PREFIX)/lib/lv2/carla.lv2/
+	$(LINK) $(LIBDIR)/carla/styles $(DESTDIR)$(LIBDIR)/lv2/carla.lv2/
 
 	# Link resources for lv2 plugin
-	rm -rf $(DESTDIR)$(PREFIX)/lib/lv2/carla.lv2/resources
-	$(LINK) $(PREFIX)/share/carla/resources/ $(DESTDIR)$(PREFIX)/lib/lv2/carla.lv2/
+	rm -rf $(DESTDIR)$(LIBDIR)/lv2/carla.lv2/resources
+	$(LINK) $(PREFIX)/share/carla/resources/ $(DESTDIR)$(LIBDIR)/lv2/carla.lv2/
 
 	# --------------------------------------------------------------------------------------------------------------------
 
@@ -620,25 +622,25 @@
 ifeq ($(HAVE_X11),true)
 ifeq ($(DEFAULT_QT),4)
 	# Install vst plugin
-	install -d $(DESTDIR)$(PREFIX)/lib/vst/carla.vst/
+	install -d $(DESTDIR)$(LIBDIR)/vst/carla.vst/
 
 	install -m 644 \
 		bin/CarlaRack*.* \
 		bin/CarlaPatchbay*.* \
-		$(DESTDIR)$(PREFIX)/lib/vst/carla.vst/
+		$(DESTDIR)$(LIBDIR)/vst/carla.vst/
 
 	# Link binaries for vst plugin
-	@for i in $(shell find $(DESTDIR)$(PREFIX)/lib/carla/ -maxdepth 1 -type f -exec basename {} ';'); do \
-		$(LINK) $(PREFIX)/lib/carla/$$i $(DESTDIR)$(PREFIX)/lib/vst/carla.vst/; \
+	@for i in $(shell find $(DESTDIR)$(LIBDIR)/carla/ -maxdepth 1 -type f -exec basename {} ';'); do \
+		$(LINK) $(LIBDIR)/carla/$$i $(DESTDIR)$(LIBDIR)/vst/carla.vst/; \
 	done
-	rm -f $(DESTDIR)$(PREFIX)/lib/vst/carla.vst/libcarla_standalone2.*
+	rm -f $(DESTDIR)$(LIBDIR)/vst/carla.vst/libcarla_standalone2.*
 
 	# Link styles for vst plugin
-	$(LINK) $(PREFIX)/lib/carla/styles $(DESTDIR)$(PREFIX)/lib/vst/carla.vst/
+	$(LINK) $(LIBDIR)/carla/styles $(DESTDIR)$(LIBDIR)/vst/carla.vst/
 
 	# Link resources for vst plugin
-	rm -rf $(DESTDIR)$(PREFIX)/lib/vst/carla.vst/resources
-	$(LINK) $(PREFIX)/share/carla/resources/ $(DESTDIR)$(PREFIX)/lib/vst/carla.vst/
+	rm -rf $(DESTDIR)$(LIBDIR)/vst/carla.vst/resources
+	$(LINK) $(PREFIX)/share/carla/resources/ $(DESTDIR)$(LIBDIR)/vst/carla.vst/
 endif
 endif
 endif
@@ -647,7 +649,7 @@
 
 uninstall:
 	rm -f $(DESTDIR)$(PREFIX)/bin/carla*
-	rm -f $(DESTDIR)$(PREFIX)/lib/pkgconfig/carla-standalone.pc
+	rm -f $(DESTDIR)$(LIBDIR)/pkgconfig/carla-standalone.pc
 	rm -f $(DESTDIR)$(PREFIX)/share/applications/carla.desktop
 	rm -f $(DESTDIR)$(PREFIX)/share/applications/carla-control.desktop
 	rm -f $(DESTDIR)$(PREFIX)/share/icons/hicolor/*/apps/carla.png
@@ -656,9 +658,9 @@
 	rm -f $(DESTDIR)$(PREFIX)/share/icons/hicolor/scalable/apps/carla-control.svg
 	rm -f $(DESTDIR)$(PREFIX)/share/mime/packages/carla.xml
 	rm -rf $(DESTDIR)$(PREFIX)/include/carla/
-	rm -rf $(DESTDIR)$(PREFIX)/lib/carla/
-	rm -rf $(DESTDIR)$(PREFIX)/lib/lv2/carla.lv2/
-	rm -rf $(DESTDIR)$(PREFIX)/lib/vst/carla.vst/
+	rm -rf $(DESTDIR)$(LIBDIR)/carla/
+	rm -rf $(DESTDIR)$(LIBDIR)/lv2/carla.lv2/
+	rm -rf $(DESTDIR)$(LIBDIR)/vst/carla.vst/
 	rm -rf $(DESTDIR)$(PREFIX)/share/carla/
 
 # ----------------------------------------------------------------------------------------------------------------------------
diff -dur -x '*~' Carla-1.9.6.orig/data/carla Carla-1.9.6/data/carla
--- Carla-1.9.6.orig/data/carla	2015-03-12 14:09:14.000000000 +0100
+++ Carla-1.9.6/data/carla	2016-11-19 13:45:49.448810610 +0100
@@ -11,5 +11,5 @@
 fi
 
 INSTALL_PREFIX="X-PREFIX-X"
-export PATH="$INSTALL_PREFIX"/lib/carla:$PATH
+export PATH="X-LIBDIR-X/carla:$PATH"
 exec $PYTHON "$INSTALL_PREFIX"/share/carla/carla --with-appname="$0" --with-libprefix="$INSTALL_PREFIX" "$@"
diff -dur -x '*~' Carla-1.9.6.orig/data/carla-database Carla-1.9.6/data/carla-database
--- Carla-1.9.6.orig/data/carla-database	2015-03-12 14:09:14.000000000 +0100
+++ Carla-1.9.6/data/carla-database	2016-11-19 13:45:49.448810610 +0100
@@ -7,5 +7,5 @@
 fi
 
 INSTALL_PREFIX="X-PREFIX-X"
-export PATH="$INSTALL_PREFIX"/lib/carla:$PATH
+export PATH="X-LIBDIR-X/carla:$PATH"
 exec $PYTHON "$INSTALL_PREFIX"/share/carla/carla_database.py --with-libprefix="$INSTALL_PREFIX" "$@"
diff -dur -x '*~' Carla-1.9.6.orig/data/carla-patchbay Carla-1.9.6/data/carla-patchbay
--- Carla-1.9.6.orig/data/carla-patchbay	2015-03-12 14:09:14.000000000 +0100
+++ Carla-1.9.6/data/carla-patchbay	2016-11-19 13:45:49.448810610 +0100
@@ -11,5 +11,5 @@
 fi
 
 INSTALL_PREFIX="X-PREFIX-X"
-export PATH="$INSTALL_PREFIX"/lib/carla:$PATH
+export PATH="X-LIBDIR-X/carla:$PATH"
 exec $PYTHON "$INSTALL_PREFIX"/share/carla/carla-patchbay --with-appname="$0" --with-libprefix="$INSTALL_PREFIX" "$@"
diff -dur -x '*~' Carla-1.9.6.orig/data/carla-rack Carla-1.9.6/data/carla-rack
--- Carla-1.9.6.orig/data/carla-rack	2015-03-12 14:09:14.000000000 +0100
+++ Carla-1.9.6/data/carla-rack	2016-11-19 13:45:49.448810610 +0100
@@ -11,5 +11,5 @@
 fi
 
 INSTALL_PREFIX="X-PREFIX-X"
-export PATH="$INSTALL_PREFIX"/lib/carla:$PATH
+export PATH="X-LIBDIR-X/carla:$PATH"
 exec $PYTHON "$INSTALL_PREFIX"/share/carla/carla-rack --with-appname="$0" --with-libprefix="$INSTALL_PREFIX" "$@"
diff -dur -x '*~' Carla-1.9.6.orig/data/carla-settings Carla-1.9.6/data/carla-settings
--- Carla-1.9.6.orig/data/carla-settings	2015-03-12 14:09:14.000000000 +0100
+++ Carla-1.9.6/data/carla-settings	2016-11-19 13:45:49.448810610 +0100
@@ -7,5 +7,5 @@
 fi
 
 INSTALL_PREFIX="X-PREFIX-X"
-export PATH="$INSTALL_PREFIX"/lib/carla:$PATH
+export PATH="X-LIBDIR-X/carla:$PATH"
 exec $PYTHON "$INSTALL_PREFIX"/share/carla/carla_settings.py --with-libprefix="$INSTALL_PREFIX" "$@"
diff -dur -x '*~' Carla-1.9.6.orig/data/carla-single Carla-1.9.6/data/carla-single
--- Carla-1.9.6.orig/data/carla-single	2015-03-12 14:09:14.000000000 +0100
+++ Carla-1.9.6/data/carla-single	2016-11-19 13:45:49.448810610 +0100
@@ -27,7 +27,7 @@
 
 INSTALL_PREFIX = "X-PREFIX-X"
 
-CARLA_LIBDIR = os.path.join(INSTALL_PREFIX, "lib", "carla")
+CARLA_LIBDIR = os.path.join("X-LIBDIR-X", "carla")
 CARLA_RESDIR = os.path.join(INSTALL_PREFIX, "share", "carla", "resources")
 
 # --------------------------------------------------------------------------------------------------------
diff -dur -x '*~' Carla-1.9.6.orig/data/carla-standalone.pc Carla-1.9.6/data/carla-standalone.pc
--- Carla-1.9.6.orig/data/carla-standalone.pc	2015-03-12 14:09:14.000000000 +0100
+++ Carla-1.9.6/data/carla-standalone.pc	2016-11-19 13:45:49.448810610 +0100
@@ -1,6 +1,6 @@
 prefix=X-PREFIX-X
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib/carla
+libdir=X-LIBDIR-X/carla
 includedir=${prefix}/include/carla
 
 Name: carla-standalone
diff -dur -x '*~' Carla-1.9.6.orig/data/todo/carla-control Carla-1.9.6/data/todo/carla-control
--- Carla-1.9.6.orig/data/todo/carla-control	2015-03-12 14:09:14.000000000 +0100
+++ Carla-1.9.6/data/todo/carla-control	2016-11-19 13:45:49.448810610 +0100
@@ -7,5 +7,5 @@
 fi
 
 INSTALL_PREFIX="X-PREFIX-X"
-export PATH="$INSTALL_PREFIX"/lib/carla:$PATH
+export PATH="X-LIBDIR-X/carla:$PATH"
 exec $PYTHON "$INSTALL_PREFIX"/share/carla/carla_control.py --with-appname="$0" --with-libprefix="$INSTALL_PREFIX" "$@"
diff -dur -x '*~' Carla-1.9.6.orig/source/carla_shared.py Carla-1.9.6/source/carla_shared.py
--- Carla-1.9.6.orig/source/carla_shared.py	2015-03-12 14:09:14.000000000 +0100
+++ Carla-1.9.6/source/carla_shared.py	2016-11-19 13:46:47.227483196 +0100
@@ -375,23 +375,33 @@
     splitter = ":"
 
     DEFAULT_LADSPA_PATH  = HOME + "/.ladspa"
+    DEFAULT_LADSPA_PATH += ":/usr/lib64/ladspa"
     DEFAULT_LADSPA_PATH += ":/usr/lib/ladspa"
+    DEFAULT_LADSPA_PATH += ":/usr/libx32/ladspa"
     DEFAULT_LADSPA_PATH += ":/usr/local/lib/ladspa"
 
     DEFAULT_DSSI_PATH    = HOME + "/.dssi"
+    DEFAULT_DSSI_PATH   += ":/usr/lib64/dssi"
     DEFAULT_DSSI_PATH   += ":/usr/lib/dssi"
+    DEFAULT_DSSI_PATH   += ":/usr/libx32/dssi"
     DEFAULT_DSSI_PATH   += ":/usr/local/lib/dssi"
 
     DEFAULT_LV2_PATH     = HOME + "/.lv2"
+    DEFAULT_LV2_PATH    += ":/usr/lib64/lv2"
     DEFAULT_LV2_PATH    += ":/usr/lib/lv2"
+    DEFAULT_LV2_PATH    += ":/usr/libx32/lv2"
     DEFAULT_LV2_PATH    += ":/usr/local/lib/lv2"
 
     DEFAULT_VST2_PATH    = HOME + "/.vst"
+    DEFAULT_VST2_PATH   += ":/usr/lib64/vst"
     DEFAULT_VST2_PATH   += ":/usr/lib/vst"
+    DEFAULT_VST2_PATH   += ":/usr/libx32/vst"
     DEFAULT_VST2_PATH   += ":/usr/local/lib/vst"
 
     DEFAULT_VST3_PATH    = HOME + "/.vst3"
+    DEFAULT_VST3_PATH   += ":/usr/lib64/vst3"
     DEFAULT_VST3_PATH   += ":/usr/lib/vst3"
+    DEFAULT_VST3_PATH   += ":/usr/libx32/vst3"
     DEFAULT_VST3_PATH   += ":/usr/local/lib/vst3"
 
     DEFAULT_GIG_PATH     = HOME + "/.sounds/gig"
@@ -540,9 +550,10 @@
 def getPaths(libPrefix = None):
     CWDl = CWD.lower()
 
+    pathBinaries  = os.path.join("X-LIBDIR-X", "carla")
+
     # standalone, installed system-wide linux
     if libPrefix is not None:
-        pathBinaries  = os.path.join(libPrefix, "lib", "carla")
         pathResources = os.path.join(libPrefix, "share", "carla", "resources")
 
     # standalone, local source
diff -dur -x '*~' Carla-1.9.6.orig/source/carla_widgets.py Carla-1.9.6/source/carla_widgets.py
--- Carla-1.9.6.orig/source/carla_widgets.py	2015-03-12 14:09:14.000000000 +0100
+++ Carla-1.9.6/source/carla_widgets.py	2016-11-19 13:45:49.448810610 +0100
@@ -1577,7 +1577,7 @@
     loadHostSettings(host)
 
     host.engine_init("JACK", "Carla-Widgets")
-    host.add_plugin(BINARY_NATIVE, PLUGIN_DSSI, "/usr/lib/dssi/karplong.so", "karplong", "karplong", 0, None, 0x0)
+    host.add_plugin(BINARY_NATIVE, PLUGIN_DSSI, "X-LIBDIR-X/dssi/karplong.so", "karplong", "karplong", 0, None, 0x0)
     host.set_active(0, True)
 
     gui1 = CarlaAboutW(None, host)
diff -dur -x '*~' Carla-1.9.6.orig/source/modules/juce_audio_processors/format_types/juce_LADSPAPluginFormat.cpp Carla-1.9.6/source/modules/juce_audio_processors/format_types/juce_LADSPAPluginFormat.cpp
--- Carla-1.9.6.orig/source/modules/juce_audio_processors/format_types/juce_LADSPAPluginFormat.cpp	2015-03-12 14:09:14.000000000 +0100
+++ Carla-1.9.6/source/modules/juce_audio_processors/format_types/juce_LADSPAPluginFormat.cpp	2016-11-19 13:45:49.448810610 +0100
@@ -696,7 +696,7 @@
 FileSearchPath LADSPAPluginFormat::getDefaultLocationsToSearch()
 {
     return FileSearchPath (SystemStats::getEnvironmentVariable ("LADSPA_PATH",
-                                                                "/usr/lib/ladspa;/usr/local/lib/ladspa;~/.ladspa")
+                                                                "/usr/lib64/ladspa;/usr/lib/ladspa;/usr/libx32/ladspa;/usr/local/lib/ladspa;~/.ladspa")
                              .replace (":", ";"));
 }
 
diff -dur -x '*~' Carla-1.9.6.orig/source/modules/juce_audio_processors/format_types/juce_VSTPluginFormat.cpp Carla-1.9.6/source/modules/juce_audio_processors/format_types/juce_VSTPluginFormat.cpp
--- Carla-1.9.6.orig/source/modules/juce_audio_processors/format_types/juce_VSTPluginFormat.cpp	2015-03-12 14:09:14.000000000 +0100
+++ Carla-1.9.6/source/modules/juce_audio_processors/format_types/juce_VSTPluginFormat.cpp	2016-11-19 13:45:49.448810610 +0100
@@ -2818,7 +2818,7 @@
     return FileSearchPath ("~/Library/Audio/Plug-Ins/VST;/Library/Audio/Plug-Ins/VST");
    #elif JUCE_LINUX
     return FileSearchPath (SystemStats::getEnvironmentVariable ("VST_PATH",
-                                                                "/usr/lib/vst;/usr/local/lib/vst;~/.vst")
+                                                                "/usr/lib64/vst;/usr/lib/vst;/usr/libx32/vst;/usr/local/lib/vst;~/.vst")
                              .replace (":", ";"));
    #elif JUCE_WINDOWS
     const String programFiles (File::getSpecialLocation (File::globalApplicationsDirectory).getFullPathName());
diff -dur -x '*~' Carla-1.9.6.orig/source/modules/lilv/config/lilv_config.h Carla-1.9.6/source/modules/lilv/config/lilv_config.h
--- Carla-1.9.6.orig/source/modules/lilv/config/lilv_config.h	2015-03-12 14:09:14.000000000 +0100
+++ Carla-1.9.6/source/modules/lilv/config/lilv_config.h	2016-11-19 13:45:49.448810610 +0100
@@ -29,7 +29,7 @@
 #elif defined(__WIN32__)
  #define LILV_DEFAULT_LV2_PATH "%APPDATA%\\LV2;%COMMONPROGRAMFILES%\\LV2"
 #else
- #define LILV_DEFAULT_LV2_PATH "~/.lv2:/usr/lib/lv2:/usr/local/lib/lv2"
+ #define LILV_DEFAULT_LV2_PATH "~/.lv2:/usr/lib64/lv2:/usr/lib/lv2:/usr/libx32/lv2:/usr/local/lib/lv2"
 #endif
 
 #endif /* _LILV_CONFIG_H_ */
