diff -dur -x '*~' Carla-1.9.8.orig/Makefile Carla-1.9.8/Makefile
--- Carla-1.9.8.orig/Makefile	2018-01-22 22:05:35.000000000 +0100
+++ Carla-1.9.8/Makefile	2018-02-03 21:14:30.198488505 +0100
@@ -375,7 +375,7 @@
 	install -d $(DESTDIR)$(BINDIR)
 ifeq ($(LINUX),true)
 	install -d $(DESTDIR)$(LIBDIR)/carla/jack
-	install -d $(DESTDIR)$(LIBDIR)/python3/dist-packages
+	install -d $(DESTDIR)$(PYPKGDIR)
 else
 	install -d $(DESTDIR)$(LIBDIR)/carla
 endif
@@ -447,7 +447,7 @@
 	install -m 644 \
 		source/carla_backend.py \
 		source/carla_utils.py \
-		$(DESTDIR)$(LIBDIR)/python3/dist-packages
+		$(DESTDIR)$(PYPKGDIR)
 endif
 
 	# Install headers
